common/shared.c

```diff
@@ -1398,28 +1404,27 @@ static int ft_inject_progress(uint64_t total)
 ssize_t ft_post_tx(struct fid_ep *ep, fi_addr_t fi_addr, size_t size,
 		uint64_t data, struct fi_context* ctx)
 {
+	void *mr_desc = (ft_check_mr_local_flag(fi) && mr) ? fi_mr_desc(mr) : NULL;
```

Comment:
This could be made global and set once after the memory region is registered. That could avoid repeating the code in every function.

Reply:
Nice point. The PR has been updated accordingly